### ISSUE TITLE
CAMS-CSM1-0-tasmax.yaml fails cleaning validation for ssp370, ssp245, ssp126, ssp585

### ISSUE THREAD

**Issue by brews** [ClimateImpactLab/downscaleCMIP6#346]
CAMS-CSM1-0-tasmax.yaml fails cleaning validation for historical, ssp245, ssp126, ssp585 in https://argo.cildc6.org/workflows/default/e2e-tasmax-jobs-56tb5

Blocks progress on run #201.

From ssp370 log:
```
Validating gs://clean-b1dbca25/cmip6/ScenarioMIP/CAMS/CAMS-CSM1-0/ssp370/r2i1p1f1/day/tasmax/gn/v20200720.zarr
Traceback (most recent call last):
  File "/argo/staging/script", line 18, in <module>
    _test_timesteps(ds, data_type, time_period)
  File "/opt/dodola/dodola/core.py", line 645, in _test_timesteps
    assert (
AssertionError: projection cmip6 file is missing timesteps, only has 35040
```

log from ssp245:
```
Validating gs://clean-b1dbca25/cmip6/ScenarioMIP/CAMS/CAMS-CSM1-0/ssp245/r2i1p1f1/day/tasmax/gn/v20200720.zarr
Traceback (most recent call last):
  File "/argo/staging/script", line 18, in <module>
    _test_timesteps(ds, data_type, time_period)
  File "/opt/dodola/dodola/core.py", line 645, in _test_timesteps
    assert (
AssertionError: projection cmip6 file is missing timesteps, only has 35040
```
log from ssp126:
```
Validating gs://clean-b1dbca25/cmip6/ScenarioMIP/CAMS/CAMS-CSM1-0/ssp126/r2i1p1f1/day/tasmax/gn/v20191106.zarr
Traceback (most recent call last):
  File "/argo/staging/script", line 18, in <module>
    _test_timesteps(ds, data_type, time_period)
  File "/opt/dodola/dodola/core.py", line 645, in _test_timesteps
    assert (
AssertionError: projection cmip6 file is missing timesteps, only has 35040
```

log from ssp585:
```
Validating gs://clean-b1dbca25/cmip6/ScenarioMIP/CAMS/CAMS-CSM1-0/ssp585/r2i1p1f1/day/tasmax/gn/v20191106.zarr
Traceback (most recent call last):
  File "/argo/staging/script", line 18, in <module>
    _test_timesteps(ds, data_type, time_period)
  File "/opt/dodola/dodola/core.py", line 645, in _test_timesteps
    assert (
AssertionError: projection cmip6 file is missing timesteps, only has 35040
```

I'd say the inputs are missing timesteps (ha ha).

**Comment by brews** [comment]
Heads up to @dgergel

**Comment by dgergel** [comment]
@brews here we actually _are_ missing timesteps, the input zarrs only go through 2099. I don't think this requires any updates for the workflow but the validation code can check the last year of the data, and if it is not 2100, expect a smaller number of timesteps (and flag this for us). Sound reasonable?

**Comment by brews** [comment]
@dgergel Hmmm. We don't want time steps with **less** than 2015 - 2099 though, right? 15 years of data, for example, is no good. This is a case where you want validation to fail. So then maybe we want a value >= the number of days from 2015 - 2099 on a noleap calendar?

The other thing is that we run currently run QDM from 2015 to 2100. Always. This is used for that rolling window loop. Do you always want this to end at 2100 regardless of what's available in the data or do you want it to go until 2099 if the input data only goes to 2099?

**Comment by dgergel** [comment]
Yeah if we have less than 2015 - 2099 we want to know. There was a model in CMIP5 where that was the case, so not to say we'd eliminate it for that reason but we def want to know. 

I don't see a reason to go past 2099 if we only have input data that goes to 2099

**Comment by brews** [comment]
Closed with PR #365
